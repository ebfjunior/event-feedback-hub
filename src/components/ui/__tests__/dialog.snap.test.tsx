import React from 'react';
import { render } from '@testing-library/react';
import { Dialog, DialogContent, DialogHeader, DialogTitle } from '../dialog';

vi.mock('lucide-react', async () => {
	const actual = await vi.importActual<any>('lucide-react');
	return { ...actual, XIcon: (p: any) => <svg data-icon="x" {...p} /> };
});

describe('ui/Dialog (snapshot)', () => {
	it('renders portal content when open', () => {
		render(
			<Dialog open>
				<DialogContent>
					<DialogHeader>
						<DialogTitle>Title</DialogTitle>
					</DialogHeader>
				</DialogContent>
			</Dialog>,
		);
		expect(document.body).toMatchSnapshot();
	});
});


