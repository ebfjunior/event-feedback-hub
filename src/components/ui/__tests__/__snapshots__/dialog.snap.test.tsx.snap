// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/Dialog (snapshot) > renders portal content when open 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  />
  <div
    aria-hidden="true"
    class="data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 fixed inset-0 z-50 bg-black/50"
    data-aria-hidden="true"
    data-slot="dialog-overlay"
    data-state="open"
    style="pointer-events: auto;"
  />
  <div
    aria-describedby="radix-«r2»"
    aria-labelledby="radix-«r1»"
    class="bg-background data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 fixed top-[50%] left-[50%] z-50 grid w-full max-w-[calc(100%-2rem)] translate-x-[-50%] translate-y-[-50%] gap-4 rounded-lg border p-6 shadow-lg duration-200 sm:max-w-lg"
    data-slot="dialog-content"
    data-state="open"
    id="radix-«r0»"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
  >
    <div
      class="flex flex-col gap-2 text-center sm:text-left"
      data-slot="dialog-header"
    >
      <h2
        class="text-lg leading-none font-semibold"
        data-slot="dialog-title"
        id="radix-«r1»"
      >
        Title
      </h2>
    </div>
    <button
      class="ring-offset-background focus:ring-ring data-[state=open]:bg-accent data-[state=open]:text-muted-foreground absolute top-4 right-4 rounded-xs opacity-70 transition-opacity hover:opacity-100 focus:ring-2 focus:ring-offset-2 focus:outline-hidden disabled:pointer-events-none [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
      data-slot="dialog-close"
      type="button"
    >
      <svg
        data-icon="x"
      />
      <span
        class="sr-only"
      >
        Close
      </span>
    </button>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
